gap/digraph.gi

```diff
@@ -1631,6 +1636,40 @@ function(filt, n, p)
     return DigraphNC(adjacencyList);
 end);
 
+InstallMethod(RandomDigraphCons,
+"for IsStronglyConnectedDigraph, a positive integer, and a float",
+[IsStronglyConnectedDigraph, IsPosInt, IsFloat],
+function(filt, n, p)
+  local d, adjMatrix, stronglyConnectedComponents,
+  scc_a, scc_b, i, random_u, random_v;
+
+  # strong connected digraph must be at least connected
+  d := RandomDigraph(IsConnectedDigraph, n, p);
+
+  stronglyConnectedComponents := DigraphStronglyConnectedComponents(d);
+
+  adjMatrix := AdjacencyMatrixMutableCopy(d);
+
+  for i in [1 .. Size(stronglyConnectedComponents.comps) - 1] do
+      scc_a := stronglyConnectedComponents.comps[i];
+      scc_b := stronglyConnectedComponents.comps[i + 1];
+
+      # add a connection from u to v
+      random_u := Random(scc_a);
+      random_v := Random(scc_b);
+
+      adjMatrix[random_u][random_v] := 1;
+
+      # get a different u and v and add edge in the reverse direction
+      random_u := Random(scc_b);
+      random_v := Random(scc_a);
+
+      adjMatrix[random_u][random_v] := 1;
```

Comment:
I think we could add fewer edges here by using a cycle instead of double-edges.

You're adding edges
$$v_1 \leftrightarrow v_2 \leftrightarrow v_3 \leftrightarrow \ldots \leftrightarrow v_n$$

but you could add edges

$$v_1 \to v_2 \to v_3 \to \ldots \to v_n \to v_1$$

thus adding $n$ edges instead of $2n-2$ edges.  Not a big deal, but should be a simple fix?